kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,255 @@
+    {
+        "Parameters" : {
+            "materials_filename" : "please specify the file to be opened"
+        }
+    }  )"
+    );
+
+    rParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    // read json string in materials file, create Parameters
+    const std::string& materials_filename = rParameters["Parameters"]["materials_filename"].GetString();
+    std::ifstream infile(materials_filename);
+    KRATOS_ERROR_IF_NOT(infile.good()) << "The masterials file cannot be found: " << materials_filename << std::endl;
+    std::stringstream buffer;
+    buffer << infile.rdbuf();
+    Parameters materials(buffer.str());
+
+    GetPropertyBlock(materials);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+ReadMaterialsUtility::ReadMaterialsUtility(
+    const std::string& rParametersName,
+    Model& rModel
+    ) : mrModel(rModel)
+{
+    // Receive json string with materials properties, create Parameters
+    Parameters materials(rParametersName);
+
+    GetPropertyBlock(materials);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::GetPropertyBlock(Parameters materials)
+{
+    KRATOS_INFO("Read materials") << "Started" << std::endl;
+    for (auto i = 0; i < materials["properties"].size(); ++i) {
+        Parameters material = materials["properties"].GetArrayItem(i);
+        AssignPropertyBlock(material);
+    }
+    KRATOS_INFO("Read materials") << "Finished" << std::endl;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::string CleanVariableName(std::string);
+std::string CleanVariableName(std::string line){
+    std::stringstream ss(line);
+    std::string variable_name;
+    while (std::getline(ss, variable_name, '.')){}
+    return variable_name;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::AssignPropertyBlock(Parameters data)
+{
+    // Get the properties for the specified model part.
+    ModelPart& model_part = mrModel.GetModelPart(data["model_part_name"].GetString());
+    const IndexType property_id = data["properties_id"].GetInt();
+    const IndexType mesh_id = 0;
+    Properties::Pointer p_prop = model_part.pGetProperties(property_id, mesh_id);
+
+    KRATOS_INFO_IF("::[Reading materials process]:: Property", (data["Material"]["Variables"].begin() - data["Material"]["Variables"].end()) > 0 && p_prop->HasVariables()) << std::to_string(property_id) << "already has variables." << std::endl;
```

Comment:
Here it should be .end() - .begin(). Anyways, we already tried this method to test for previous Variables and Tables and it doesn't work. I couldn't find a simple way to test it.